Make LoTAS components translatable

ScribbleTAS opened the pull request:
- [Fabric] Fixed crash when using F3+Esc
- [Fabric]  Added translation loading via LoTASLanguageManager
- Added 1.20
# TODO
## Fabric
- [x] LoTAS Gui
- [x] Keybinds
- [x] AiManip
- [x] Config
- [x] DragonManip
- [x] InfoHud
- [x] Loadstate
- [x] Spawnmanip
- [x] DropManip
   - [x] Deadbush
   - [x] Glowstone
   - [x] Gravel
   - [x] Leaf
   - [x] Ore
   - [x] Plants
   - [x] SeaLantern
   - [x] Drowned
   - [x] Fish
   - [x] Monster
   - [x] Nether
   - [x] Passive
   - [x] Zombie
   - [x] Bartering
## Forge
- [x] LoTAS Gui
- [x] Keybinds
- [x] AiManip
- [x] Config
- [x] DragonManip
- [x] InfoHud
- [x] Loadstate
- [x] Spawnmanip
- [x] DropManip
   - [x] Deadbush
   - [x] Glowstone
   - [x] Gravel
   - [x] Leaf
   - [x] Ore
   - [x] Plants
   - [x] SeaLantern
   - [x] Monster
   - [x] Nether
   - [x] Passive
   - [x] Zombie

ScribbleTAS committed:
[Fabric] Fixed crash when using F3+Esc

Reported by AlPie

George417 commented:
nice

ScribbleTAS committed:
[Fabric] Updated loom

- Fixed error in console when respawning

ScribbleTAS committed:
Added the ability to load lang files

- [Fabric] Added LoTASLanguageManager for loading resources from the jar or from Minecrafts resource manager
- [Forge] Added lang files and pack.mcmeta because sometimes forge is a bit more convenient

ScribbleTAS commented on LoTAS-Fabric/src/main/java/de/pfannekuchen/lotas/core/utils/KeybindsUtils.java:
```diff
@@ -108,12 +108,11 @@ public static void tickKeyEvent() {
 		// Autostrafe auto rotation handling
 		if (wasPressed != holdStrafeKeybind.isDown() && wasPressed == true) {
-			KeyMapping.set(Minecraft.getInstance().options.keyRight.getDefaultKey(), false);
+			Minecraft mc = Minecraft.getInstance();
```
Fixed some resource leak warnings

ScribbleTAS committed:
[Fabric] LoTASIngameGui, AiManip, Keybinds

ScribbleTAS committed:
[Fabric] DragonManip, Config, DropManip

- Rewrote BarteingManip a bit

ScribbleTAS committed:
[Fabric] InfoHud, LoadstateScreen

ScribbleTAS committed:
[Fabric] Even more translations

PancakeTAS commented on LoTAS-Fabric/build.gradle:
```diff
@@ -1,6 +1,6 @@
 import groovy.json.JsonOutput
 
-version = "2.1.0"
+version = "2.1.2"
```
wat

PancakeTAS commented on LoTAS-Fabric/src/main/java/de/pfannekuchen/lotas/core/utils/LoTASLanguageManager.java:
you nailed implementing the i18n _standard_

ScribbleTAS replied:
I don't know if this is sarcastic or a compliment

ScribbleTAS committed:
Fixed things

ScribbleTAS committed:
[WIP] Added 1.20

Mojang why are you doing this? Does the game need a new rendering engine every 4 versions?????

ScribbleTAS committed:
Added 1.20

ScribbleTAS committed:
Fixed small errors when updating to 1.20

ScribbleTAS committed:
Fixed wrong mixin in MixinTickrateChangerAchievements

PancakeTAS reviewed (approved):
awesome, i will never read through this code again

ScribbleTAS committed:
[Fabric 1.16] Fixed wrong preprocessor statement breaking rendering

ScribbleTAS committed:
[Fabric] Added I18n for SpawnManipMod

ScribbleTAS commented:
> awesome, i will never read through this code again

Never say never

ScribbleTAS committed:
[Forge] I18n for Keybinds, DragonManip, InfoHud

ScribbleTAS committed:
[Forge] I18n for DropManip, SpawnManip and more
- [Forge] Fixed weird coloring of textures when hovering over buttons in dropmanip
- [Fabric] Misc translation fixes
